components/automate-cli/cmd/chef-automate/verify.go

```diff
@@ -204,126 +197,281 @@ func (v *verifySystemdCreateFlow) runVerifySystemdCreateCmd(cmd *cobra.Command,
 
 func verifyCmdFunc(flagsObj *verifyCmdFlags) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		c := verifyCmdFlow{
-			Verification:      &verification.VerificationModule{},
-			A2HARBFileExist:   isA2HARBFileExist(),
-			ManagedServicesOn: isManagedServicesOn(),
+		defaultLevel := "info"
+		if flagsObj.debug {
+			defaultLevel = "debug"
 		}
+
+		log, err := logger.NewLogger("text", defaultLevel)
+		if err != nil {
+			return err
+		}
+
+		createSystemdServiceWithBinary, err := verifysystemdcreate.NewCreateSystemdService(
+			verifysystemdcreate.NewSystemdCreateUtilsImpl(),
+			BINARY_DESTINATION_FOLDER,
+			SYSTEMD_PATH,
+			flagsObj.debug,
+			writer,
+		)
+		if err != nil {
+			return err
+		}
+
+		c := NewVerifyCmdFlow(httputils.NewClient(log), createSystemdServiceWithBinary, verifysystemdcreate.NewSystemdCreateUtilsImpl(), config.NewHaDeployConfig(), sshutils.NewSSHUtilWithCommandExecutor(sshutils.NewSshClient(), log, command.NewExecExecutor()), writer)
 		return c.runVerifyCmd(cmd, args, flagsObj)
 	}
 }
 
 func (v *verifyCmdFlow) runVerifyCmd(cmd *cobra.Command, args []string, flagsObj *verifyCmdFlags) error {
-	var configPath = ""
-	if len(args) > 0 {
-		configPath = args[0]
+	var configPath string
+
+	// TODO : config flag is optional for now. Need to handle the default config path
+	if len(strings.TrimSpace(flagsObj.config)) > 0 {
+		configPath = flagsObj.config
 	}
 
-	switch true {
-	case flagsObj.haAWSProvision:
-		err := v.verifyHaAWSProvision(configPath)
-		return err
-	case flagsObj.haAWSManagedProvision:
-		err := v.verifyHaAWSManagedProvision(configPath)
-		return err
-	case flagsObj.haAWSDeploy:
-		err := v.verifyHaAWSDeploy(configPath)
-		return err
-	case flagsObj.haAWSManagedDeploy:
-		err := v.verifyHaAWSManagedDeploy(configPath)
-		return err
-	case flagsObj.standaloneDeploy:
-		err := v.verifyStandaloneDeploy(configPath)
-		return err
-	case flagsObj.haOnpremDeploy:
-		err := v.verifyHaOnpremDeploy(configPath)
-		return err
-	case flagsObj.haOnPremAWSManagedDeploy:
-		err := v.verifyHaOnPremAWSManagedDeploy(configPath)
+	v.prettyPrint = flagsObj.prettyPrint
+
+	err := v.Config.ParseAndVerify(configPath)
+	if err != nil {
 		return err
-	case flagsObj.haOnPremCustManagedDeploy:
-		err := v.verifyHaOnPremCustManagedDeploy(configPath)
+	}
+
+	// Get config required for batch-check API call
+	batchCheckConfig := &models.Config{}
+	batchCheckConfig.PopulateWith(v.Config)
```

Comment:
This returns error, please handle the error

Reply:
done